modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/FDSPropsTransformer/FDSDataRenderers/EntryStatusDataRenderer.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import React, {useEffect, useState} from 'react';
+
+import '../../../../object_entries/ViewObjectEntries.scss';
+
+import {fetch, sub} from 'frontend-js-web';
+
+function getDisplayType(status: string) {
+	switch (status) {
+		case 'approved':
+			return 'success';
+		case 'pending':
+		case 'scheduled':
+			return 'info';
+		case 'expired':
+			return 'warning';
+		default:
+			return 'secondary';
+	}
+}
+
+export default function EntryStatusDataRenderer({
+	itemData,
+	restContextPath,
+}: {
+	itemData: ObjectEntry;
+	restContextPath: string;
+}) {
+	const [versions, setVersions] = useState<ObjectEntry[]>([]);
+
+	const displayDate = itemData.displayDate
+		? new Date(itemData.displayDate)
+		: null;
+
+	const versionNumber = itemData.systemProperties?.version?.number;
+
+	const statusLabel = itemData.status.label;
+
+	useEffect(() => {
+		async function makeFetch() {
+			const response = await fetch(
+				`${restContextPath}/${itemData.id}/versions`
+			);
+
+			const data = await response.json();
+
+			setVersions(data.items);
+		}
+
+		if (versionNumber) {
```

Comment:
If the `versionNumber` is undefined the useEffect will still create the  `makeFetch`. What do you think about use that if in the begining of the useEffect, like:

```
useEffect(() => {
    if (versionNumber) {
		async function makeFetch() {
			const response = await fetch(
				`${restContextPath}/${itemData.id}/versions`
			);

			const data = await response.json();

			setVersions(data.items);
		}
		
		makeFetch();
}
}, [versionNumber, itemData.id, restContextPath]);
```

Reply:
makes sense, just did it! I just had to change it use a function expression instead of a function declaration since that way it would violate a ESLint rule.